### PR TITLE
Add mousetrap bindings for bold, italics, and links in TextEditor

### PR DESCRIPTION
**Pull request by todtb**
#2722 

This adds mousetrap bindings for bold, italics, and links in the `TextEditor` component. The use of refs required converting `TextEditorFormatButton` from a functional component. 

Additionally, this can't really be handled in the `keyboard_shortcuts.js` file as we need to bind to an element rendered by the component rather than the default `document`. 

I'm not 100% certain this is the cleanest approach, but it seems to click-test well. Open to any suggestions to make it more idiomatic.

### CODE REVIEW

File: app/webpack/shared/components/text_editor.jsx [hunk 1]
```diff
@@ -7,12 +7,21 @@ class TextEditor extends React.Component {
...
+    this.italicButton = React.createRef();
+    this.linkButton = React.createRef();
     this.state = {
       textareaChars: 0,
       preview: false
     };
   }
 
+  componentDidMount( ) {
+    Mousetrap( this.textarea.current ).bind( 'mod+b', ( ) => { this.boldButton.button.current.click( ); } );
```
Comment (kueda) [inline app/webpack/shared/components/text_editor.jsx:20]: While this is working, Mousetrap should be imported or declared as a global in an eslint comment. Also, double-quotes only, please. React-specific eslint rules are at https://github.com/inaturalist/inaturalist/blob/master/app/webpack/.eslintrc

File: app/webpack/shared/components/text_editor.jsx [hunk 2]
```diff
@@ -43,6 +52,7 @@ class TextEditor extends React.Component {
                 label={<i className="fa fa-bold" />}
                 template={text => `**${text}**`}
                 placeholder={I18n.t( "bold_text" )}
+                ref={(button) => {this.boldButton = button;}}
```
Comment (kueda) [inline app/webpack/shared/components/text_editor.jsx:55]: Parens around `button` are extraneous and curly braces around the function definition need spaces on the inside (also in the eslint rules).

File: app/webpack/shared/components/text_editor_format_button.jsx [hunk 3]
```diff
@@ -1,63 +1,74 @@
...
+      <button
+        type="button"
+        tabIndex="-1"
+        className={className}
+        disabled={disabled}
+        title={tip}
+        aria-label={tip}
+        ref={this.button}
+        onClick={() => {
+          const {selectionStart} = textarea;
```
Comment (kueda) [inline app/webpack/shared/components/text_editor_format_button.jsx:32]: Spaces inside the braces, please.

File: app/webpack/shared/components/text_editor_format_button.jsx [hunk 4]
```diff
@@ -1,63 +1,74 @@
...
+          if (textarea.selectionStart !== undefined && textarea.selectionEnd !== undefined) {
```
Comment (kueda) [inline app/webpack/shared/components/text_editor_format_button.jsx:33]: Spaces inside parens.

File: app/webpack/shared/components/text_editor_format_button.jsx [hunk 5]
```diff
@@ -1,63 +1,74 @@
...
-    }}
-  >
-    { label }
-  </button>
-);
+class TextEditorFormatButton extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.button = React.createRef();
+  };
```
Comment (kueda) [inline app/webpack/shared/components/text_editor_format_button.jsx:8]: Class methods don't need semi-colons.

**Comment by todtb** [comment]
I tested `preventDefualt( )` on 78.0.2 (64-bit) and that should do the trick. I think that's acceptable in this case given that it is only bound to the textarea.

Not sure if we need to support IE, but it (of course) uses a slightly different approach I've left out for now.

**Comment by kueda** [comment]
Looking great. Thanks again!

**Pull request merged**